MediaManager/Plugin.MediaManager.iOS/MediaRemoteControl.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Threading.Tasks;
+using Plugin.MediaManager.Abstractions;
+using UIKit;
+
+namespace Plugin.MediaManager
+{
+    public class MediaRemoteControl : IMediaRemoteControl
+    {
+        private readonly IPlaybackController _playbackController;
+
+        private readonly TimeSpan _seekInterval = TimeSpan.FromMilliseconds(1000);
+
+        private bool _seeking;
+
+        public MediaRemoteControl(IPlaybackController playbackController)
+        {
+            _playbackController = playbackController;
+        }
+
+        public void RemoteControlReceived(UIEvent uiEvent)
+        {
+            HandleRemoteControlEvent(uiEvent.Subtype);
+        }
+
+        private void HandleRemoteControlEvent(UIEventSubtype eventType)
+        {
+            switch (eventType)
+            {
+                case UIEventSubtype.RemoteControlPlay:
+                    _playbackController.Play();
+                    break;
+
+                case UIEventSubtype.RemoteControlPause:
+                    _playbackController.Pause();
+                    break;
+
+                case UIEventSubtype.RemoteControlTogglePlayPause:
+                    _playbackController.PlayPause();
+                    break;
+
+                case UIEventSubtype.RemoteControlStop:
+                    _playbackController.Stop();
+                    break;
+
+                case UIEventSubtype.RemoteControlNextTrack:
+                    _playbackController.PlayNext();
+                    break;
+
+                case UIEventSubtype.RemoteControlPreviousTrack:
+                    _playbackController.PlayPreviousOrSeekToStart();
+                    break;
+
+                case UIEventSubtype.RemoteControlBeginSeekingForward:
+                    StartSeekingForward();
+                    break;
+
+                case UIEventSubtype.RemoteControlBeginSeekingBackward:
+                    StartSeekingBackward();
+                    break;
+
+                case UIEventSubtype.RemoteControlEndSeekingForward:
+                case UIEventSubtype.RemoteControlEndSeekingBackward:
+                    _seeking = false;
+                    break;
+            }
+        }
+
+        private void StartSeekingForward()
+        {
+            if(TryStartSeek())
+            {
+                Task.Run(() => ExecuteAndWaitLoop(_playbackController.StepForward));
```

Comment:
This is an implementation that is only used for iOS because it allows you to seek holding down the next and previous buttons. There is now way you can do this on Android or any other platform that I know of. I am generally against placing comments anywhere as I believe in the code telling the story by using intuitive naming. I can however look into the naming here as I think there is room for improvement.